File: 01/python/exercise12.py
class Payroll:
    @staticmethod
    def baseSalary(hours, rate):
        return hours * rate

    @staticmethod
    def withholdingTax(salary):
        return salary * 0.125

    @staticmethod
    def netSalary(amount, tax):
        return amount - tax


baseSalary = Payroll.baseSalary(48, 5000)
tax = Payroll.withholdingTax(baseSalary)
netSalary = Payroll.netSalary(baseSalary, tax)

print(f"El salario bruto es {baseSalary}")
print(f"La retefuente es de {tax}")
print(f"El salario neto es {netSalary}")

File: 01/csharp/exercise12.cs
using System;

class Payroll {
    public static int BaseSalary(int hours, int rate) 
    {
        return hours * rate;
    }

    public static double WithholdingTax(int salary) 
    {
        return salary * 0.125;
    } 

    public static double NetSalary(int amount, double tax) 
    {
        return amount - tax;
    }
}

class MainClass {
    public static void Main() {
        int BaseSalary = Payroll.BaseSalary(48, 5000);
        double Tax = Payroll.WithholdingTax(BaseSalary);
        double NetSalary = Payroll.NetSalary(BaseSalary, Tax);

        Console.WriteLine($"El salario bruto es {BaseSalary}");
        Console.WriteLine($"La retefuente es de {Tax}");
        Console.WriteLine($"El salario neto es {NetSalary}");
    }
}